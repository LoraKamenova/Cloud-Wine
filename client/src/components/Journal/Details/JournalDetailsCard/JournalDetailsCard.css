.jdc-content-wrapper {
    display: flex;
    flex-direction: column;
    width: 100%;
    background-color: #fff;
    background-image:
            linear-gradient(90deg, transparent 79px, #abced4 79px, #abced4 81px, transparent 81px),
            linear-gradient(#eee .1em, transparent .1em);
    background-size: 100% 1.7em;
}

.journal-details-card {
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.jdc-first-section {
    margin: 0 40px 0 100px;
}

.jdc-second-section {
    margin-bottom: 60px;
}

.jdc-heading {
    font-family: 'Caveat', cursive;
    font-size: 50px;
    text-align: center;
}

.jdc-subsection {
    display: flex;
    flex-direction: column;
}

.jdc-subheading {
    font-size: 30px;
    font-family: 'Caveat', cursive;
    text-decoration: underline;
    margin: 20px 0 10px 0px;
}

.jdc-text {
    text-align: justify;
    line-height: 1.5;
}

.jdc-image-container {
    display: flex;
    flex-direction: column;
}


.jdc-image-and-tape-wrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.jdc-subheading.image-and-tape-heading {
    margin-left: 100px;
    margin-bottom: 0;
}

.tape {
    position: relative;
    bottom: -20px;
    width: 250px;
    height: 80px;
    background-image: url("https://res.cloudinary.com/dyhmxus4n/image/upload/v1616498543/React%20Project/tape_rel617.png");
}

.left-tape {
    transform: rotate(5deg);
}

.right-tape {
    transform: rotate(-5deg);
}

.jdc-image {
    width: 550px;
    height: 350px;
    box-shadow: 0px 0px 9px -3px #000000;
    margin-bottom: 10px;
}

@media all and (max-width: 1024px) {
    .jdc-image {
        width: 350px;
        height: 200px;
    }

    .tape {
        width: 150px;
        height: 50px;
    }
}
