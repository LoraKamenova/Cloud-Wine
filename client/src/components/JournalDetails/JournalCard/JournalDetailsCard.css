
.journal-details-card {
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.journal-details-card-header {
    text-align: center;
}

.journal-details-card-heading {
    font-size: 35px;
}

.journal-details-card-image-wrapper {
    display: flex;
    justify-content: space-around;
    margin: 20px 0;
}

.journal-details-card-image {
    width: 420px;
    height: 250px;
    box-shadow: 0px 0px 11px -3px #000000;
}

.journal-details-card-text {
    padding: 20px 40px;
    text-align: justify;
}

.journal-details-card-button-wrapper {
    display: flex;
    justify-content: flex-end;
}
