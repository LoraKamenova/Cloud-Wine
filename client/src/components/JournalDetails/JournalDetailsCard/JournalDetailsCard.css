.journal-details-card {
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 20px 50px 30px 50px;
}

.journal-details-card-header {
    text-align: center;
}

.journal-details-card-heading {
    font-family: 'Caveat', cursive;
    font-size: 50px;
}

.journal-details-card-subheading {
    font-size: 30px;
    font-family: 'Caveat', cursive;
    text-decoration: underline;
    margin-bottom: 10px;
}

.journal-details-card-wrapper, .journal-details-card-image-wrapper {
    display: flex;
    flex-direction: column;
    margin-top: 30px;
}

.journal-details-card-image-container {
    display: flex;
    justify-content: space-around;
    margin: 20px 0;
}

.journal-details-card-image {
    width: 420px;
    height: 250px;
    box-shadow: 0px 0px 11px -3px #000000;
    margin: 10px;
}

.journal-details-card-text {
    text-align: justify;
}

.journal-details-card-button-wrapper {
    display: flex;
    justify-content: flex-end;
}

@media only screen and (max-width: 1024px) {
    .journal-details-card-image-wrapper {
        flex-direction: column;
        align-items: center;
    }
    .journal-details-card-button-wrapper {
        justify-content: center;
    }
}
