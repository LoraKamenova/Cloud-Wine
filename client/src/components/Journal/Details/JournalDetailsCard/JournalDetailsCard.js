import './JournalDetailsCard.css'

const JournalDetailsCard = (props) => {
    return (
        <div className="journal-details-card">

            <div className="jdc-content-wrapper">
                <section className="jdc-first-section">
                    <h1 className="jdc-heading">{props.title}</h1>

                    <div className="jdc-subsection">
                        <h3 className="jdc-subheading">Къде ходихме</h3>
                        <div className="jdc-text jdc-place">{props.place}</div>
                    </div>

                    <div className="jdc-subsection">
                        <h3 className="jdc-subheading">Кога отидохме</h3>
                        <div className="jdc-text jdc-time">{props.time}</div>
                    </div>

                    <div className="jdc-subsection">
                        <h3 className="jdc-subheading">Какво правихме</h3>
                        <div className="jdc-text jdc-description">{props.description}</div>
                    </div>
                </section>

                <section className="jdc-second-section">
                    <h3 className="jdc-subheading image-and-tape-heading">Какво снимахме</h3>
                    <div className="jdc-image-and-tape-wrapper">
                        <div className="tape left-tape"></div>
                        <img className="jdc-image" src={props.imageUrl1} alt="journal"/>
                        <div className="tape right-tape"></div>
                        <img className="jdc-image" src={props.imageUrl2} alt="journal"/>
                    </div>
                </section>
            </div>

        </div>
    );
};

export default JournalDetailsCard;
