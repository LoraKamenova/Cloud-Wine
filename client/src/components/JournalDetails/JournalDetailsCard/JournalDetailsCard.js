import './JournalDetailsCard.css'

const JournalDetailsCard = (props) => {
    return (
        <div className="journal-details-card">

            <section className="jdc-first-section">
                    <h1 className="jdc-heading">{props.title}</h1>

                <div className="jdc-subsection">
                    <h3 className="jdc-subheading">Къде ходихме</h3>
                    <div className="jdc-text">{props.place}</div>
                </div>

                <div className="jdc-subsection">
                    <h3 className="jdc-subheading">Кога отидохме</h3>
                    <div className="jdc-text">{props.time}</div>
                </div>

                <div className="jdc-subsection">
                    <h3 className="jdc-subheading">Какво правихме</h3>
                    <div className="jdc-text">{props.description}</div>
                </div>
            </section>

            <section className="jdc-second-section">
                <h3 className="jdc-subheading image-and-tape-heading">Какво снимахме</h3>
                <div className="jdc-image-and-tape-wrapper">
                    <div className="tape"></div>
                    <img className="jdc-image" src={props.imageUrl1} alt="journal"/>
                    <div className="tape"></div>
                    <img className="jdc-image" src={props.imageUrl2} alt="journal"/>
                </div>
            </section>

            <div className="jdc-button-wrapper">
                <a className="button edit-button" href="#">Редактирай<i className="fas fa-pencil-alt"></i></a>
                <a className="button delete-button" href="#">Изтрий<i className="fas fa-times"></i></a>
            </div>

        </div>
    );
};

export default JournalDetailsCard;
