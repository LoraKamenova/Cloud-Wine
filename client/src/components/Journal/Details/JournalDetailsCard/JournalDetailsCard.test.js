import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import JournalDetailsCard from './JournalDetailsCard';

describe('JournalDetailsCard Component', () => {
    it('Should display entry title', () => {
        render(
            <BrowserRouter>
                <JournalDetailsCard title="Незабравима ваканция" link="/"/>
            </BrowserRouter>
        );

        expect(document.querySelector(".jdc-heading").textContent).toBe('Незабравима ваканция');
    });

    it('Should display place', () => {
        render(
            <BrowserRouter>
                <JournalDetailsCard place="Шато Коларово" link="/"/>
            </BrowserRouter>
        );

        expect(document.querySelector(".jdc-place").textContent).toBe('Шато Коларово');
    });

    it('Should display time', () => {
        render(
            <BrowserRouter>
                <JournalDetailsCard time="лято 2020" link="/"/>
            </BrowserRouter>
        );

        expect(document.querySelector(".jdc-time").textContent).toBe('лято 2020');
    });

    it('Should display description', () => {
        render(
            <BrowserRouter>
                <JournalDetailsCard description="Това е разказ какво правихме през лятната отпуска." link="/"/>
            </BrowserRouter>
        );

        expect(document.querySelector(".jdc-description").textContent).toBe('Това е разказ какво правихме през лятната отпуска.');
    });

    it("JournalDetailsCard Component snapshot", () => {
        const { asFragment } = render(<BrowserRouter><JournalDetailsCard link="/"/></BrowserRouter>);
        expect(asFragment()).toMatchSnapshot();
    });

});
