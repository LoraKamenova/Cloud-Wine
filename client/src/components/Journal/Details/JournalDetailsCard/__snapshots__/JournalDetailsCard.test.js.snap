// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JournalDetailsCard Component JournalDetailsCard Component snapshot 1`] = `
<DocumentFragment>
  <div
    class="journal-details-card"
  >
    <div
      class="jdc-content-wrapper"
    >
      <section
        class="jdc-first-section"
      >
        <h1
          class="jdc-heading"
        />
        <div
          class="jdc-subsection"
        >
          <h3
            class="jdc-subheading"
          >
            Къде ходихме
          </h3>
          <div
            class="jdc-text jdc-place"
          />
        </div>
        <div
          class="jdc-subsection"
        >
          <h3
            class="jdc-subheading"
          >
            Кога отидохме
          </h3>
          <div
            class="jdc-text jdc-time"
          />
        </div>
        <div
          class="jdc-subsection"
        >
          <h3
            class="jdc-subheading"
          >
            Какво правихме
          </h3>
          <div
            class="jdc-text jdc-description"
          />
        </div>
      </section>
      <section
        class="jdc-second-section"
      >
        <h3
          class="jdc-subheading image-and-tape-heading"
        >
          Какво снимахме
        </h3>
        <div
          class="jdc-image-and-tape-wrapper"
        >
          <div
            class="tape left-tape"
          />
          <img
            alt="journal"
            class="jdc-image"
          />
          <div
            class="tape right-tape"
          />
          <img
            alt="journal"
            class="jdc-image"
          />
        </div>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
